docs/source/examples/word-count.rst

```diff
@@ -0,0 +1,219 @@
+    ('3', 221407)]
+
+Cluster computation with HDFS
+-----------------------------
+
+Next, we'll use ``dask`` along with the `distributed`_ with the `hdfs3`_
+libraries to count the number of words in all of the text files stored in HDFS.
+
+Copy the text data from S3 into HDFS on the cluster:
+
+.. code-block:: bash
+
+   $ hadoop distcp s3n://{AWS_SECRET_ID}:{AWS_SECRET_KEY}@blaze-data/enron-email hdfs:///tmp/enron
+
+where ``AWS_SECRET_ID`` and ``AWS_SECRET_KEY`` are valid AWS credentials.
+
+Start the ``distributed`` scheduler and workers on the cluster. Because of the
+parallelizable nature of ``bag`` operations, we can start the ``distributed``
+workers with multiple processes equal to the number of cores on the machine:
+
+.. code-block:: bash
+
+   $ dscheduler  # On the head node
+   $ dworker EXECUTOR_IP:EXECUTOR_PORT --nprocs 4 --nthreads 1  # On the compute nodes
+
+In the above ``dworker`` command, replace ``EXECUTOR_IP`` and ``EXECUTOR_PORT``
+with the IP address and port of the ``distributed`` scheduler.
+
+In Python, import the ``hdfs3`` and the ``distributed`` methods used in this
+example:
+
+.. code-block:: python
+
+   >>> import hdfs3
+   >>> from distributed import Executor, progress
+   >>> from distributed.hdfs import read_text
+
+Initalize a connection to HDFS, replacing ``NAMENODE_HOSTNAME`` and
+``NAMENODE_PORT`` with the hostname and port (default: 8020) of the HDFS
+namenode:
+
+.. code-block:: python
+
+   >>> hdfs = hdfs3.HDFileSystem('NAMENODE_HOSTNAME', port=NAMENODE_PORT)
+
+Initalize a connection to the ``distributed`` executor:
+
+.. code-block:: python
+
+   >>> e = Executor('EXECUTOR_IP:EXECUTOR_PORT')
+
+Create a ``bag`` from the text files stored in HDFS. This expression will not
+read data from HDFS until the computation is triggered:
+
+.. code-block:: python
+
+   >>> b = read_text('/tmp/enron/*/*', hdfs=hdfs)
+
+   Setting global dask scheduler to use distributed
+
+Generate a word count expression using the same ``bag`` methods as the local
+``dask`` example:
+
+.. code-block:: python
+
+   >>> wordcount = b.str.split().concat().frequencies().topk(10, lambda x: x[1])
+
+We are ready to count the number of words in all of the text files using
+``distributed`` workers. We can map the ``wordcount`` expression to a future
+that triggers the computation on the cluster.
+
+.. code-block:: python
+
+   >>> future = e.compute(wordcount)
+
+Note that the ``compute`` operation is non-blocking, and you can continue to
+work in the Python shell/notebook while the computations are running.
+
+We can check the status of the ``future`` while all of the text files are being
+processed:
+
+.. code-block:: python
+
+   >>> print(future)
+
+   <Future: status: pending, key: finalize-0f2f51e2350a886223f11e5a1a7bc948>
+
+   >>> progress(future)
+
+   [########################################] | 100% Completed |  8min  15.2s
+
+This operation required about 8 minutes to run on a cluster with three worker
```

Comment:
I added a line to the local dask section to say that the single file calculation took 7 seconds. I also added a line to the cluster section to say that the calculation took 20 minutes to run on a single machine with the same specs.